tests/components/advantage_air/test_init.py

```diff
@@ -0,0 +1,39 @@
+"""Test the Advantage Air Initialization."""
+
+from advantage_air import ApiError
+
+from homeassistant.components.advantage_air.const import DOMAIN
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from tests.components.advantage_air import (
+    TEST_SYSTEM_DATA,
+    TEST_SYSTEM_URL,
+    add_mock_config,
+)
+
+
+async def test_async_setup_entry(hass, aioclient_mock):
+    """Test a successful setup entry."""
+
+    aioclient_mock.get(
+        TEST_SYSTEM_URL,
+        text=TEST_SYSTEM_DATA,
+    )
+
+    entry = await add_mock_config(hass)
+
+    assert isinstance(
+        hass.data[DOMAIN][entry.entry_id]["coordinator"], DataUpdateCoordinator
+    )
+    assert callable(hass.data[DOMAIN][entry.entry_id]["async_change"])
```

Comment:
Please remove all assertions of integration details.
```suggestion
```

Reply:
Here we want to assert the state of the config entry.
https://github.com/home-assistant/core/blob/c8ac787c9f7be9059dfefdf937bcb270f87aa226/homeassistant/config_entries.py#L56-L67

Reply:
Thanks for the guidance here, this has been added.